### ISSUE TITLE
integration-test-profile not being considered when using quarkus-azure-functions extension

### ISSUE THREAD

**Issue by julianrschmidt** [quarkusio/quarkus#42868]
### Describe the bug

When using `@QuarkusIntegrationTest` I would like to be able to change the used profile, as it is easily possible with other quarkus projects using `quarkus.test.integration-test-profile=someprofile`. 
However, when using the `quarkus-azure-functions` extension, the used profile will always be `prod`.

### Expected behavior

Setting `quarkus.test.integration-test-profile=someprofile` inside the `application.properties` file should set the used profile to `someprofile`when running a `@QuarkusIntegrationTest`.

### Actual behavior

The used profile is `prod`.

### How to Reproduce?

1. Checkout the example project via `quarkus create app --extension=quarkus-azure-functions`
2. Add the property `quarkus.test.integration-test-profile=someprofile` to the file `application.properties`
3. Run the integration test inside the class `FunctionIT` and observe the used profile to be `prod` instead of `someprofile`.

### Output of `uname -a` or `ver`

Darwin AO-FHC6G6XH14.local 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:13:04 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6020 arm64 arm Darwin

### Output of `java -version`

openjdk version "17.0.9" 2023-10-17 LTS

### Quarkus version or git rev

3.14.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.8

### Additional information

As a workaround it is possible to execute `./mnvw verify -DskipITs=false -Dquarkus.profile=someprofile`, which will actually change the profile for the integration test.

**Comment by geoand** [comment]
Is this still an issue with the latest versions of Quarkus?

**Comment by julianrschmidt** [comment]
Yes, I could reproduce the steps above with the current version of quarkus 3.23.0.

**Comment by geoand** [comment]
Thanks for the update
